pkg/apis/pipeline/v1beta1/pipelinerun_conversion.go

```diff
@@ -208,6 +214,144 @@ func (ptrs *PipelineTaskRunSpec) convertFrom(ctx context.Context, source v1.Pipe
+	for _, st := range prs.SkippedTasks {
+		new := v1.SkippedTask{}
+		st.convertTo(ctx, &new)
+		sink.SkippedTasks = append(sink.SkippedTasks, new)
+	}
+	sink.ChildReferences = nil
+	for _, cr := range prs.ChildReferences {
+		new := v1.ChildStatusReference{}
+		cr.convertTo(ctx, &new)
+		sink.ChildReferences = append(sink.ChildReferences, new)
+	}
+	sink.FinallyStartTime = prs.FinallyStartTime
+	if prs.Provenance != nil {
+		new := v1.Provenance{}
+		prs.Provenance.convertTo(ctx, &new)
+		sink.Provenance = &new
+	}
+	return nil
+}
+
+func (prs *PipelineRunStatus) ConvertFrom(ctx context.Context, source *v1.PipelineRunStatus) error {
+	// TODO: handle duckv1beta1.Status to v1
+	prs.StartTime = source.StartTime
+	prs.CompletionTime = source.CompletionTime
+	// TaskRuns are dropped
+	prs.PipelineResults = nil
+	for _, pr := range source.Results {
+		new := PipelineRunResult{}
+		new.convertFrom(ctx, pr)
+		prs.PipelineResults = append(prs.PipelineResults, new)
+	}
+	if source.PipelineSpec != nil {
+		newPipelineSpec := PipelineSpec{}
+		err := newPipelineSpec.ConvertFrom(ctx, source.PipelineSpec)
+		if err != nil {
+			return err
+		}
+		prs.PipelineSpec = &newPipelineSpec
+	}
+	prs.SkippedTasks = nil
+	for _, st := range source.SkippedTasks {
+		new := SkippedTask{}
+		new.convertFrom(ctx, st)
+		prs.SkippedTasks = append(prs.SkippedTasks, new)
+	}
+	prs.ChildReferences = nil
+	for _, cr := range source.ChildReferences {
+		new := ChildStatusReference{}
+		new.convertFrom(ctx, cr)
+		prs.ChildReferences = append(prs.ChildReferences, new)
+	}
+	prs.FinallyStartTime = source.FinallyStartTime
+	if source.Provenance != nil {
+		new := Provenance{}
+		new.convertFrom(ctx, *source.Provenance)
+		prs.Provenance = &new
+	}
+	return nil
+}
+
+func (prr PipelineRunResult) convertTo(ctx context.Context, sink *v1.PipelineRunResult) {
+	sink.Name = prr.Name
+	newValue := v1.ParamValue{}
+	prr.Value.convertTo(ctx, &newValue)
+	sink.Value = v1.ResultValue(newValue)
+}
+
+func (prr *PipelineRunResult) convertFrom(ctx context.Context, source v1.PipelineRunResult) {
+	prr.Name = source.Name
+	newValue := ParamValue{}
+	newValue.convertFrom(ctx, source.Value)
+	prr.Value = newValue
+}
+
+func (st SkippedTask) convertTo(ctx context.Context, sink *v1.SkippedTask) {
+	sink.Name = st.Name
+	sink.Reason = v1.SkippingReason(st.Reason)
+	sink.WhenExpressions = nil
+	for _, we := range st.WhenExpressions {
+		new := v1.WhenExpression{}
+		we.convertTo(ctx, &new)
+		sink.WhenExpressions = append(sink.WhenExpressions, new)
+	}
+}
+
+func (st *SkippedTask) convertFrom(ctx context.Context, source v1.SkippedTask) {
+	st.Name = source.Name
+	st.Reason = SkippingReason(source.Reason)
+	st.WhenExpressions = nil
+	for _, we := range source.WhenExpressions {
+		new := WhenExpression{}
+		new.convertFrom(ctx, we)
+		st.WhenExpressions = append(st.WhenExpressions, new)
+	}
+}
+
+func (csr ChildStatusReference) convertTo(ctx context.Context, sink *v1.ChildStatusReference) {
```

Comment:
Do we need to convert the TypeMeta for all the fields or only for ChildStatusReference?

Reply:
Naively I would have say "for all" 🤔, this might be a "miss" for `task.Typemeta`

Reply:
I'd actually say we shouldn't convert the `TypeMeta` of the actual `ChildStatusReference`s, since that's identifying information about the resource being referenced, and that resource has already been created by the time it ends up in `ChildReferences`.

Reply:
So I think I may be thinking of "convert" differently - what I mean is that we should _preserve_ the `TypeMeta`s in `ChildStatusReference` when we convert the `PipelineRunStatus`, as I mentioned elsewhere on this PR, but somehow I got confused by this thread and repeated myself in a confusing manner. =)

Reply:
It seems like the term "convert" is used somehow differently 😅. 

As far as I understand, what vincent means, which is the idea I agreed, is we don't need to touch `Typemeta` of primitive CRDs (task, taskrun, pipeline, pipelinerun), but we need to **manually deal with** (which is called `convert` here, can be done by copying over etc) the `TypeMeta` that appear in other places. We are pretty much aligned if I'm not mistaken.